### PR TITLE
fix(oracle): Add that we support Oracle 19c

### PR DESCRIPTION
**Pull request by danila-m**
* Modify installation guidance

### CODE REVIEW

**Review by pabloalonso — APPROVED** [review]

**Review by jjargot — CHANGES_REQUESTED** [review]
! Thank you for this update !

It is require to always mention the real full Oracle version used, such as Oracle 12c (12.2.0.x.y).
Once installed, connect to the server as Oracle user:
$ sqlplus / as sysdba
select * from v$version;

Q1- Would you mind to use the release version in the output of the SQL query above plus the marketing one all together?

The customers using the Oracle 12c (12.2.0.x.y) needs to run the initxa.sql and xaview.sql scripts, which are creating both the XA Subprograms and the XA views: AKA Oracle XA Library.

It is wise to implement 2 sections: 
 -1- Oracle 12c (12.2.0.x.y) -> current working procedure
 -2- Oracle 19c (? ? ? ? ? ?) -> new section

Q2- What do you think?


--
-- Oracle 19c (? ? ? ? ? ?) 

The official Oracle documentation pages regarding the Oracle XA Library have not changed.
That is why we think at support that the scripts are still needed.

I noticed that the XA Subprogram may be already installed with Enterprise Edition but on **Windows Only**

Q3- What do you think?

If both initxa.sql and xaview.sql scripts are delivered with the Oracle 19c installation, support suggests to let the customer run them: in other word, do not remove them from the procedure (plz)

Q4- What do you think?

If you agree with this, only one section will be needed for all Oracle versions, where the customers are instructed to tun the scripts.

--
-- A valuable check (AKA Feedback from customer site)

One customer did run several times the xa scripts to install the Oracle XA Library on the Oracle 12 (12.2.0.x.y) , but the changes were somehow rolled back. This was causing errors with bonita 7.9.

Christophe from PS has provided the procedure to test that the Oracle XA Libray is properly installed
'
Sur la base de données (Oracle) :

**Depuis le user bonita :**

Démarrer la transaction : call dbms_xa.xa_start(xid => dbms_xa_id(1234), flag => dbms_xa.tmnoflags) ;

vérifier la transaction (si ko, ce n’est pas correctement configuré) : dbms_xa.xa_ok et call dbms_xa.xa_getlastoer()

Suspendre la transaction : call dbms_xa.xa_end(xid => dbms_xa_id(1234), flag =>  dbms_xa.tmsuspend)

vérifier dbms_xa.xa_ok et call dbms_xa.xa_getlastoer() pour vérifier si tout va bien ou si il y a déjà une anomalie

vérifier qu’on retrouve la transaction : select state,  flags,  coupling from  gv$global_transaction where globalid = dbms_xa_xid(1234).gtrid;

**Depuis le user bdm :**

récupérer la transaction : call dbms_xa.xa_start(xid => dbms_xa_id(1234),   flag =>  dbms_xa.tmresume)

vérifier la transaction (si ko, ce n’est pas correctement configuré) : dbms_xa.xa_ok et call dbms_xa.xa_getlastoer()

Fermer la transaction : call dbms_xa.xa_end(xid => dbms_xa_id(1234), flag =>  dbms_xa.tmsuccess)

vérifier dbms_xa.xa_ok et call dbms_xa.xa_getlastoer() pour vérifier si tout va bien ou si il y a déjà une anomalie
'

It would be a great enhencement to include this check in the documentation.

Q5- What do you think?

File: md/database-configuration.md [hunk 1]
```diff
@@ -46,8 +46,9 @@ Bonita provides out of the box the Jdbc drivers for H2, PostgreSQL, Microsoft SQ
 
 #### Oracle Database
 
-*Warning*: Bonita 7.9.+ is compatible with Oracle 12.2.0.x only. Make sure you use the driver that works with your Oracle database version, make sure you check your driver's Checksum. 
-Download the driver compatible with Oracle 12.2.0.1: [ojdbc8.jar](https://www.oracle.com/database/technologies/jdbc-ucp-122-downloads.html) ( make sure it is the offical driver by checking the SHA1 Checksum: 60f439fd01536508df32658d0a416c49ac6f07fb )
+*Warning*: Bonita 7.9.+ is compatible with Oracle 12.2.0.x and Oracle 19c (19.3.0.0). Make sure you use the driver that works with your Oracle database version, make sure you check your driver's Checksum. 
+The driver compatible with Oracle 12.2.0.x can be found here: [ojdbc8.jar](https://www.oracle.com/database/technologies/jdbc-ucp-122-downloads.html) ( make sure it is the offical driver by checking the SHA1 Checksum: 60f439fd01536508df32658d0a416c49ac6f07fb )
```
Comment (pabloalonso) [inline md/database-configuration.md]: Same driver for 12.2 and 19?

File: md/hardware-and-software-requirements.md [hunk 2]
```diff
@@ -34,7 +34,7 @@ Software required for Bonita Platform (Bonita Engine and Bonita Portal).
 | MySQL                              | 8.0.14 and higher in the 8.0.x line (see note 3) |
 | PostgreSQL                         | 11.2 and higher in the 11.x line                 |
 | SQL Server                         | 2016                                             |
-| Oracle                             | 12c (12.2.0.x) (see note 4)                      |
+| Oracle                             | 12c (12.2.0.x), 19c (19.3.0.0.0) (see note 4)                      |
```
Comment (pabloalonso) [inline md/hardware-and-software-requirements.md:37]: Why 19.3 and before 19.0?

File: md/database-configuration.md (hunk 1 above)
Comment (tbouffard) [reply md/database-configuration.md]: why not using the driver available on maven central?
https://medium.com/oracledevs/oracle-jdbc-drivers-on-maven-central-64fcf724d8b
can we embed it into bonita?

Comment (pabloalonso) [reply md/database-configuration.md]: not for 7.9 and 7.10 as there are 2 different drivers. It can be in 7.11

Comment (danila-m) [reply md/database-configuration.md]: We can, and probably will in 7.11. We did not in 7.10.

Comment (marielle-spiteri) [reply md/database-configuration.md]: If you say that Oracle 19c (19.3.0.0) is supported, could you please provide the link to the driver for 19.3.0.0 and the checksum. It's a recurring source of giant problems, really it is a pita, and it must be over explained in the doc. Thanks for your understanding.

Comment (danila-m) [reply md/database-configuration.md]: I did in this very PR ? 
If the resulting documentation https://documentation.bonitasoft.com/bonita/7.10/database-configuration is not clear somewhat please let me know.

Comment (marielle-spiteri) [reply md/database-configuration.md]: Was the driver 967c0b1a2d5b1435324de34a9b8018d294f8f47b fully tested with both 12.2 and 12.3? Again, we've been down that road before, we know that each version of oracle needs the related specific driver version. so if there is one driver for 12.2 and one driver for 12.3, and an other one for both version, all cases should tested and documented in my opinion.

**Pull request merged**